code/main.lua

```diff
@@ -8,91 +8,195 @@ activity.setTitle('鲲')
+		print("ERROR: DIVIDED BY 0")
+		print("隐藏成就计数器："..tostring(found).."/7。")
+		achievementCounter = 0
+	end
+	--1
+	if notChanging == 1 then
+		print("您就不能换一个数据嘛！我生气了！我要把答案搞成错误的！")
+		result.text="340"
+		found = found + 1
+		print("隐藏成就达成："..tostring(found).."/7")
+		notChanging = notChanging + 1
+	end
+	--3
+	if fortyTwo == 2 then
+		found = found + 1
+		print("隐藏成就达成："..tostring(found).."/7")
+		adder1.text="SIX"
+		运算符.text="MULTIPLIED BY"
+		adder2.text="NINE"
+		fortyTwo = fortyTwo + 1
+	end
+	--6
+	if calcAndCalc == 50 then
 		print("这是您第50次计算。那么，象征性给您成就半个！")
 		found=found+0.5
 		print("隐藏成就达成："..tostring(found).."/7")
+		calcAndCalc = calcAndCalc + 1 
 	end
-	if ultimate==1 and adder2.text=="42" then
-		found = found + 1
+	--7
+	if knowingWhatKunWants > 0 and knowingWhatKunWants < 11 then
+		print("说，您调戏这个计算器多久了？那么熟练？")
+		knowingWhatKunWants = knowingWhatKunWants + 1
+	end
+	if knowingWhatKunWants > 10 then
+		print("不给不给不给"..string.rep("！",knowingWhatKunWants))
+		knowingWhatKunWants = knowingWhatKunWants + 1
+	end
+	if knowingWhatKunWants == 20 then
+		found=found+1
+		print("好的，好的。你赢了。不就是成就嘛。给你加一个。")
 		print("隐藏成就达成："..tostring(found).."/7")
-		adder1.text="YESH!"
-		运算符.text="It's 42!"
-		ultimate=0
-	else
-		if modechange < 20 then modechange = 0 end
-		if mode==0 then
-			result.text=tostring(tonumber(adder1.text)+tonumber(adder2.text))
-			if adder1.text == "20" and adder2.text == "10" and time==1 then
-				print("您就不能换一个数据嘛！我生气了！我要把答案搞成错误的！")
-				result.text="340"
-				found = found + 1
-			print("隐藏成就达成："..tostring(found).."/7")
-			end
-		end
-		if mode==1 then
-		result.text=tostring(tonumber(adder1.text)-tonumber(adder2.text))
-		end
-		if mode==2 then
-			result.text=tostring(tonumber(adder1.text)*tonumber(adder2.text))
-			if result.text=="340" then
-				if knowing == 0 then
-					print("说，您调戏这个计算器多久了？那么熟练？")
-				end
-				count = count + 1
-				if count==10 then
-					found=found+1
-					print("好的，好的。你赢了。不就是成就嘛。给你加一个。")
-					print("隐藏成就达成："..tostring(found).."/7")
-				else
-					if count > 10 then
-					print("再按也没有了。")
-					else
-					print("不给不给不给"..string.rep("！",count))
-					end
-				end
-			end
-		end
-		if mode==3 then
-			if adder2.text~="0" then
-				result.text=tostring(tonumber(adder1.text)/tonumber(adder2.text))
-			else
-				if error_catch.checked then
-					print("ERROR: DIVIDED BY 0")
-					print("隐藏成就计数器："..tostring(found).."/7。")
-				end
-			end
-		end
+		knowingWhatKunWants = knowingWhatKunWants +1
 	end
-end
 
+	--coming soon!
+
+
+	--end
+		--nothing to do
+
+
+	--easter eggs
+
+end
 
 config.onClick=function(changeMode)
+	--the main part,because this is really a calculator
+
 	mode = (mode + 1) % 4
-	modechange = modechange+1
 	运算符.text = string.sub("+-*/",mode+1,mode+1)
 	print("现在运算符变成了 "..string.sub("+-*/",mode+1,mode+1))
-	if modechange==20 then
+
+
+	--Ifs
+
+	--2
+	if hittingModes < 20 then
+		hittingModes = hittingModes + 1
+	end
+	--4
+	if cancelling > 0 then
```

Comment:
The `cancelling` achievement can be repeatedly acquired now. Fix this (similarly as the above).

Reply:
*Actually I do not expect the five 'Cancel's to be consecutive...*

Reply:
Oh yes, the corrected code:
`if cancelling > 0 and cancelling < 5 then`

Reply:
Done.

Reply:
All right.

Reply:
Done.